pymatgen/core/interface.py

```diff
@@ -22,18 +24,21 @@
 from pymatgen.core.structure import Structure
 from pymatgen.core.surface import Slab
 from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
+from pymatgen.util.typing import Tuple3Ints
 
 if TYPE_CHECKING:
     from collections.abc import Sequence
-    from typing import Any
+    from typing import Any, Callable
 
-    from numpy.typing import ArrayLike
+    from numpy.typing import ArrayLike, NDArray
     from typing_extensions import Self
 
-    from pymatgen.util.typing import CompositionLike, Vector3D
+    from pymatgen.core import Element
+    from pymatgen.util.typing import CompositionLike, Matrix3D, MillerIndex, Tuple3Floats, Vector3D
+
+Tuple4Ints = tuple[int, int, int, int]
```

Comment:
I put it here because it doesn't seem to be needed by other code, `tuple[int, int, int, int]` doesn't seem to be commonly used. But we could relocate it if needed

Reply:
ah, didn't see that. i would relocate right away